docs/ADR-Resize-Channel-Security.md

```diff
@@ -0,0 +1,130 @@
+# Architectural Decision Record: Channel Resize Security
+
+## Status
+Proposed
+
+## Date
+2025-03-06
+
+## Context
+The Nitrolite system supports payment channels across multiple blockchain networks through the Clearnode server. Users can resize their channels to adjust the allocated funds. Currently, the resize flow works as follows:
+
+1. User calls the Clearnode API with a resize request
+2. Clearnode validates the request and user balance
+3. Clearnode generates and signs a new state with RESIZE intent
+4. Clearnode returns the signed state to the user
+5. User signs the state and submits it to the smart contract
+
+This process has revealed a vulnerability: users can double-spend their funds by requesting resizes across multiple chains that collectively exceed their total balance. Since the server signs states that remain valid indefinitely and doesn't track pending state commitments across chains, a user could:
+
+1. Deposit 5 USDC on Chain 1 and 5 USDC on Chain 2 (total balance of 10 USDC)
+2. Request a resize of Channel 1 from 5 to 10 USDC (server signs this state)
+3. Request a resize of Channel 2 from 5 to 10 USDC (server signs this state too)
+4. Submit both states to their respective blockchains
+5. Result: User has committed 20 USDC despite only having 10 USDC in their balance
+
+## Decision Drivers
+* Security: Prevent double-spending across multiple chains
+* User Experience: Maintain a seamless resize process
+* System Integrity: Ensure balance accounting remains accurate
+* Technical Feasibility: Implement with minimal changes to existing contracts
+* Operational Overhead: Consider the impact on server operations
+
+## Considered Options
+
+### Option 1: Timeout-Based Fund Locking
+* Lock funds in the database when a resize is requested
+* Set a timeout to release locks if not confirmed on-chain
+* Challenge: The smart contract has no notion of state expiration, so users could still submit expired states
```

Comment:
We can add expiration logic to the Adjudicator contract - this would not change Custody and is rather an expected development. However, such feature requires adding timestamp to appdata.